src/scan.l

```diff
@@ -682,7 +682,19 @@ F	RET_CONST(zeek::val_mgr->False()->Ref())
 
 "0x"{HEX}+	RET_CONST(zeek::val_mgr->Count(static_cast<zeek_uint_t>(strtoull(yytext, 0, 16))).release())
 
-({H}".")+{HTLD}		RET_CONST(zeek::detail::dns_mgr->LookupHost(yytext).release())
+({H}".")+{HTLD} {
+	zeek::TableValPtr result = zeek::detail::dns_mgr->empty_addr_set();
+
+	if ( ! zeek::detail::parse_only ) {
+		const char *allow = getenv("ZEEK_DNS_ALLOW_SCRIPT_RESOLVE");
+		if ( ! allow || std::string(allow) == "0" )
+			zeek::reporter->Error("Prevented DNS lookup of '%s' at script parsing time. Set the ZEEK_DNS_ALLOW_SCRIPT_RESOLVE environment variable to enable this feature.", yytext);
```

Comment:
Nit: wondering whether we shouldn't hidfe `ZEEK_DNS_ALLOW_SCRIPT_RESOLVE` in the docs, and instead suggest one of the `lookup_hostname*` functions.

Reply:
I doubt this is used enough. It's also unclear if `lookup_hostname()` is the right tool (hard to use because it's async) - it might be input or config framework instead.